File: 02._Data_Files/parser.py
import csv
import json
import xml.etree.ElementTree as ET
import yaml

# CSV read and parse ---
# CSV file path
csv_path = 'me.csv'

# List to collect parsed CSV data
csv_data = []

with open(csv_path, newline='') as csvfile:
    reader = csv.reader(csvfile)
    for row in reader:
        csv_data.append(row)

# Print the collected CSV data
print("Data parsed from CSV-File:", csv_data)

# JSON read and parse
# JSON file path
json_path = 'me.json'

with open(json_path, 'r') as jsonfile:
    json_data = json.load(jsonfile)

# Print the parsed JSON data
print("Data parsed from JSON-File:", json_data)

# XML read and parse
# XML file path
xml_path = 'me.xml'

xml_data = ET.parse(xml_path).getroot()

# Convert XML data to dictionary
xml_dict = {}
for child in xml_data:
    if len(child) > 0:
        xml_dict[child.tag] = {}
        for item in child:
            xml_dict[child.tag][item.tag] = item.text
    else:
        xml_dict[child.tag] = child.text

# Print the parsed XML data
print("Data parsed from XML-File:", xml_dict)

# YAML read and parse
# YAML file path
yaml_path = 'me.yaml'

with open(yaml_path, 'r') as yamlfile:
    yaml_data = yaml.safe_load(yamlfile)

# Print the parsed YAML data
print("Data parsed from YAML-File:", yaml_data)

File: 02._Data_Files/parser.js
// Imports
import fs from 'fs';
import csvParser from 'csv-parser';
import xml2js from 'xml2js';
import yaml from 'js-yaml';

// Paths
const csvPath = 'me.csv';
const jsonPath = 'me.json';
const xmlPath = 'me.xml';
const yamlPath = 'me.yaml';
const txtPath = 'me.txt';

export function readAndParseCSV() {
  return new Promise((resolve, reject) => {
    const csvData = [];
    fs.createReadStream(csvPath)
      .pipe(csvParser())
      .on('data', (row) => {
        csvData.push(row);
      })
      .on('end', () => {
        resolve(csvData);
      })
      .on('error', (err) => {
        reject(err);
      });
  });
}

export function readAndParseJSON() {
  try {
    const data = fs.readFileSync(jsonPath, 'utf8');
    const jsonData = JSON.parse(data);
    return jsonData;
  } catch (error) {
    throw error;
  }
}

export function readAndParseXML() {
  try {
    const data = fs.readFileSync(xmlPath, 'utf8');
    let result;
    xml2js.parseString(data, { explicitArray: false }, (parseErr, parsedResult) => {
      if (parseErr) {
        throw parseErr;
      }
      result = parsedResult;
    });
    return result;
  } catch (error) {
    throw error;
  }
}

export function readAndParseYAML() {
  try {
    const data = fs.readFileSync(yamlPath, 'utf8');
    const yamlData = yaml.load(data);
    return yamlData;
  } catch (error) {
    throw error;
  }
}

export function readAndParseTxt() {
  try {
    const data = fs.readFileSync(txtPath, 'utf8');
    const parsedData = yaml.load(data);
    return parsedData;
  } catch (error) {
    throw error;
  }
}